rviz_common/src/rviz_common/add_display_dialog.cpp

```diff
@@ -446,8 +452,10 @@ void DisplayTypeTree::fillTree(Factory * factory)
   }
 }
 
-TopicDisplayWidget::TopicDisplayWidget(const std::string & node_name)
-: node_name_(node_name)
+TopicDisplayWidget::TopicDisplayWidget(
+  const std::string & node_name,
+  std::unique_ptr<rviz_common::ros_integration::RosNodeAbstractionIface> node_abstraction)
+: node_name_(node_name), node_abstraction_(std::move(node_abstraction))
```

Comment:
If you're going to pass in a ros node abstraction, just get the name from that rather than also passing the node name along and storing it.